gunpowder/nodes/generic_train.py

```diff
@@ -117,7 +117,11 @@ def setup(self):
     def prepare(self, request):
         deps = BatchRequest()
         for key in self.inputs.values():
-            deps[key] = request[key]
+            if key in self.array_specs:
+                deps[key] = self.array_specs[key].copy()
```

Comment:
This means the same roi will be requested at every iteration. This would probably be fine if the `Train` node is downstream of a random location, but won't always do the right thing.
To get this to work right, the train nodes are going to need to take some sort of reference request similar to the scan node [here](https://github.com/funkey/gunpowder/blob/e523b49ca846a9fd46ab6fc0dd1040cc4a4d53b4/gunpowder/nodes/scan.py#L28). This would allow the `Train` nodes to calculate any needed input roi's based a partially complete set of outputs requested.